### PR TITLE
Add pre-built indexes for Cellranger, Cellranger-arc, simpleaf and simpleaf txp2gene

### PR DESCRIPTION
**Pull request by FelixKrueger**
As outlined in #371, we would like to enable the (re-)use of pre-computed index files from a supplied genome configuration file (such as `igenomes.config`). This fairly minimal should result in a considerable reduction in compute time and cost, thus substantially reducing a run's carbon footprint.  

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/scrnaseq/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/scrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [X] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/scrnaseq.nf [hunk 1]
```diff
@@ -37,9 +37,14 @@ workflow SCRNASEQ {
...
-    params.gtf = getGenomeAttribute('gtf')
-    params.star_index = getGenomeAttribute('star')
+    // collect paths from genome attributes file (e.g. iGenomes.config; optional)
+    params.fasta            = getGenomeAttribute('fasta')
+    params.gtf              = getGenomeAttribute('gtf')
+    params.star_index       = getGenomeAttribute('star')
+    params.salmon_index     = getGenomeAttribute('simpleaf')
+    params.txp2gene         = getGenomeAttribute('simpleaf_tx2pgene')
+    params.cellranger_index = getGenomeAttribute('cellranger')
+    params.cellranger_index = getGenomeAttribute('cellrangerarc')
```
Comment (grst) [inline workflows/scrnaseq.nf:47]: Wouldn't this always override with the `cellrangerarc` attribute?

Comment (FelixKrueger) [reply workflows/scrnaseq.nf]: Well spotted. How about we make this overriding conditional?

```nextflow
    // Make cellranger-arc index overwriting conditional
    if (params.aligner == "cellrangerarc") {
        params.cellranger_index = getGenomeAttribute('cellrangerarc')
    }
```

Comment (wzheng0520) [reply workflows/scrnaseq.nf]: Or how about in this way?
```
// Make cellranger or cellranger-arc index overwriting conditional
if (params.aligner in ["cellranger", "cellrangermulti"]){
        params.cellranger_index = getGenomeAttribute('cellranger')
    } else if (params.aligner == "cellrangerarc"){
        params.cellranger_index = getGenomeAttribute('cellrangerarc')
    }
    ch_cellranger_index = params.cellranger_index ? file(params.cellranger_index) : []
```

Comment (FelixKrueger) [reply workflows/scrnaseq.nf]: agreed, this looks cleaner. I have added a new commit.

**Review by grst — APPROVED** [review]

**Pull request merged**